wpilibc/src/main/native/cpp/PIDController.cpp

```diff
@@ -131,54 +131,86 @@ PIDController::~PIDController() {
  * This should only be called by the Notifier.
  */
 void PIDController::Calculate() {
-  bool enabled;
-  PIDSource* pidInput;
-  PIDOutput* pidOutput;
+  if (m_pidInput == nullptr) return;
+  if (m_pidOutput == nullptr) return;
```

Comment:
Doesn't this introduce a race condition?  You're accessing m_pidInput without a lock held.

Reply:
If we're only checking the pointer itself, there is no race condition. m_pidInput and m_pidOutput are only changed in the constructor before the Notifier is started. We may want to synchronize access to PIDSource::PIDGet() though because the PIDSource objects don't synchronize access to themselves.